jobapi/tokens.go

```diff
@@ -0,0 +1,23 @@
+package jobapi
+
+import (
+	"crypto/rand"
+	"fmt"
+	"math/big"
+)
+
+const letters = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ"
+
+func generateToken(length int) (string, error) {
+	lettersLen := len(letters)
+	bigLettersLen := big.NewInt(int64(lettersLen))
+	b := make([]byte, length)
+	for i := range b {
+		bigRand, err := rand.Int(rand.Reader, bigLettersLen)
+		if err != nil {
+			return "", fmt.Errorf("generating crypto-random big.Int: %w", err)
+		}
+		b[i] = letters[bigRand.Int64()%int64(lettersLen)] // mod letterslen just in case rand.Int returns a number larger than lettersLen, don't want to have a panic
+	}
+	return string(b), nil
+}
```

Comment:
This potentially feels like i'm getting a little bit too fancy with all the `big.Int`s and the cryptorandoms and such. happy to make this a slightly more normal random function if it makes sense.

Reply:
What about something like

```go
func generateToken(minBits int) (string, error) {
    b := make([]byte, (minBits+7)/8)
    rand.Read(b)
    return base64.URLEncoding.EncodeToString(b), nil
}
```